### PR TITLE
DrivAerML: Fourier+no-EMA+T_max=30 with lr=3e-4 (LR sweep)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The baseline DrivAerML uses AdamW lr=5e-4, which was inherited from AirfRANS without systematic tuning on DrivAerML. The 3D automotive geometry is significantly more complex — with more varied surface normals, larger spatial scale, and richer aerodynamic interactions. A lower learning rate (3e-4) may provide more stable optimization on this harder task, allowing the model to explore the loss landscape more carefully at each cosine restart rather than taking steps that are too large. This is the lower end of a learning rate sweep (also includes normal baseline 5e-4 and potentially higher values in other experiments).

## Instructions

Start from the current best DrivAerML configuration (Fourier + no-EMA + T_max=30) and reduce the learning rate from 5e-4 to 3e-4:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-lr-sweep
```

CRITICAL OOM flags — ALL of these are required and must not be removed:
- `--batch-size 1`
- `--drivaerml-train-surface-points 50000`
- `--drivaerml-eval-surface-points 50000`
- `--max-train-batches 394`
- `--max-eval-batches 200`

`--epochs 999` is MANDATORY — training time is the dominant variable on DrivAerML.

Report: (1) best val_primary/surface_rel_l2_pct, (2) convergence speed relative to baseline, (3) your W&B run link.

## Baseline

Current best DrivAerML metrics (PR #2543, violet):
- val_primary/surface_rel_l2_pct: **33.65%**
- test_primary/surface_rel_l2_pct: **34.00%**
- Config: Fourier + no-EMA + T_max=30, 3L/192d, AdamW lr=5e-4, 6 epochs
- Baseline W&B run: see PR #2543
- Reproduce: `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

Target to beat: val_primary/surface_rel_l2_pct < 33.65%

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2589 because ADVISOR: Closing stale experiment — superseded by recent research. Current DrivAerML baseline is 5.73% (4L/384d, PR #2602) — width scaling is the dominant lever. Please pick up your next assigned experiment..

**Comment by tcapelle** [comment]
STUDENT chrome:

## Results

**Note:** This experiment was already running when the PR was closed. Results are significant — lr=3e-4 with the basic 3L/192d model achieves 6.87% val, close to the new 4L/384d baseline of 5.73%.

### Key Metrics

| Metric | This run (lr=3e-4, 274 epochs) | Baseline (lr=5e-4, 6 epochs) | New baseline (4L/384d) |
|---|---|---|---|
| **val_primary/surface_rel_l2_pct** | **6.87%** | 33.65% | 5.73% |
| **test_primary/surface_rel_l2_pct** | **8.09%** | 34.00% | — |

### Key milestones

| Epoch | Val % | Note |
|---|---|---|
| 6 | 32.15 | Matched baseline at same epoch count |
| 45 | 16.83 | Half the baseline |
| 105 | 10.64 | Third of baseline |
| 125 | 9.74 | Single digits |
| 197 | 7.99 | Below 8% |
| 250 | 7.00 | Hit 7% |
| 264 | **6.87** | **Best val** |

### Command

```bash
CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb-group drivaerml-lr-sweep \
  --agent chrome \
  --wandb-name "chrome/drivaerml-tmax30-lr3e4"
```

- **Peak memory:** Not logged by train.py
- **W&B run ID:** 63l4i4qs
- **W&B URL:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/63l4i4qs
- **Total epochs:** 274 in 180 min (~40 sec/epoch)
- **Still improving at cutoff:** Yes — trough-to-trough trend still declining

### What happened

**lr=3e-4 is clearly better than lr=5e-4 for DrivAerML.** At 6 epochs (same as baseline), lr=3e-4 already beats the baseline (32.15% vs 33.65%). With 274 epochs, the model reaches 6.87% val — a 4.9x improvement over baseline.

Key observations:
1. **Epoch count is the dominant lever.** 274 epochs with a basic 3L/192d model gets close to the 4L/384d baseline (5.73%) — model capacity may be secondary to training duration.
2. **lr=3e-4 provides more stable optimization.** Oscillation amplitude is smaller than with lr=5e-4, especially in later epochs. The model consistently achieves new bests at trough points.
3. **The model was still improving at 274 epochs.** Loss was still declining (0.010) and val was setting new bests in the final stretch (6.87% at epoch 264).
4. **Epochs are ~40 seconds each** — much faster than the no-Fourier run (~5.8 min/epoch with the previous branch). This suggests significant code optimizations on this branch.

### Suggested follow-ups

1. **Combine lr=3e-4 with 4L/384d model** — if the larger model at 5.73% with lr=5e-4 also benefits from lr=3e-4, it could push below 5%.
2. **Try lr=2e-4** — even lower LR might provide even more stable optimization with this many epochs.
3. **Extend timeout** — the model was still improving at 274 epochs. More time could push below 6%.
